### ISSUE TITLE
Edge case with multiple identical values

### ISSUE THREAD

**Issue by ecton** [nu11ptr/assert_unordered#1]
These cases currently do not panic:

```rust
    assert_unordered::assert_eq_unordered!(vec![1, 1], vec![1]);
    assert_unordered::assert_eq_unordered!(vec![1, 1, 2], vec![1, 2, 2]);
    assert_unordered::assert_eq_unordered_set!(vec![1, 1], vec![1]);
    assert_unordered::assert_eq_unordered_set!(vec![1, 1, 2], vec![1, 2, 2]);
```

I like the nice error messages these asserts produce -- great idea!

**Comment by nu11ptr** [comment]
Nice find. This was a quick and dirty so honestly never occurred to me. Maybe just add a length comparison as well? Will need to think if more sophisticated handling should be done other than that.

**Comment by nu11ptr** [comment]
I was just thinking, the `assert_eq_unordered_set` is just a "by design" since it loads the values in a set, so your bottom two examples should not panic. The `assert_eq_unordered` macro therefore actually does have different functionality and we can fix this for that one. More documentation (and examples) needed either way.

**Comment by nu11ptr** [comment]
Fixed in 0.2.0 where possible, limitation noted where not possible, and a 3rd macro added that sorts and works in all use cases (at the expense of requiring `Ord`).
